src/conn/conn_cache_pool.c

```diff
@@ -453,22 +468,41 @@ __cache_pool_assess(WT_SESSION_IMPL *session, uint64_t *phighest)
 			continue;
 		cache = entry->cache;
 		++entries;
-		new = cache->bytes_read;
+		/* Copy the value out so it doesn't change underneath us */
+		tmp = cache->bytes_read;
 		/* Handle wrapping of eviction requests. */
-		if (new >= cache->cp_saved_read)
-			cache->cp_current_read = new - cache->cp_saved_read;
+		if (tmp >= cache->cp_saved_read)
+			cache->cp_pass_read = tmp - cache->cp_saved_read;
 		else
-			cache->cp_current_read = new;
-		cache->cp_saved_read = new;
-		if (cache->cp_current_read > highest)
-			highest = cache->cp_current_read;
+			cache->cp_pass_read = tmp;
```

Comment:
This *might be losing some information – when wrapping we use tmp directly. (means if you go from int_max-100 to 2, you will count tmp as 2 not 102). Am I missing something?